intercept/src/intercept.h

```diff
@@ -1947,10 +1947,15 @@ inline uint64_t CLIntercept::getEnqueueCounter() const
 
 inline uint64_t CLIntercept::incrementEnqueueCounter()
 {
+    uint64_t enqueueCounter = m_EnqueueCounter.load();
+    if( enqueueCounter == m_Config.StopOnEnqueueCount ) {
```

Comment:
I think it would be best to check if the enqueue counter is greater than or equal to the stop counter, otherwise we could get unlucky in some multi-threaded applications and two threads could increment the enqueue counter past the stop counter.

```suggestion
    if( enqueueCounter >= m_Config.StopOnEnqueueCount ) {
```